packages/agora-rtc-react/src/hooks/client.ts

```diff
@@ -194,17 +195,38 @@ export type FetchArgs = (() => Promise<JoinOptions>) | JoinOptions;
  * @param ready
  * @param client
  */
-export function useJoin(fetchArgs: FetchArgs, ready = true, client?: IAgoraRTCClient | null): void {
+export function useJoin(
+  fetchArgs: FetchArgs,
+  ready = true,
+  client?: IAgoraRTCClient | null,
+): { data: UID; joinComplete: boolean; error: AgoraRTCError | null } {
   const resolvedClient = useRTCClient(client);
+  const [joinComplete, setJoinComplete] = useState(false);
+  const [joinResult, setJoinResult] = useState<UID>(0);
+  const [error, setError] = useState<AgoraRTCError | null>(null);
+  const isUnmountRef = useIsUnmounted();
 
   useAsyncEffect(async () => {
     if (ready && resolvedClient) {
       try {
+        if (!isUnmountRef.current) {
+          setJoinComplete(false);
+        }
         const { appid, channel, token, uid } =
           typeof fetchArgs === "function" ? await fetchArgs() : fetchArgs;
-        await resolvedClient.join(appid, channel, token, uid);
-      } catch (error) {
-        console.error(error);
+        const result = await resolvedClient.join(appid, channel, token, uid);
+        if (!isUnmountRef.current) {
+          setJoinResult(result);
+          setError(null);
+          setJoinComplete(true);
+        }
+      } catch (err) {
+        console.error(err);
+        if (!isUnmountRef.current) {
+          setJoinResult(0);
+          setError(err as AgoraRTCError);
+          setJoinComplete(true);
```

Comment:
on catch shouldn't we set `joinComplete` as false, as join didn't succeed and we got an error?